issue_16/main.js

```diff
@@ -0,0 +1,52 @@
+const ul = document.getElementById('js-ul');
+const endpoint = 'https://api.json-generator.com/templates/5-2NhpTqkaV4/data?access_token=mihb6ib9irdyvy1iwbqg804flfdpm6kxdwtoum6x';
+
+function createNewElements(tagName, className) {
+    const newElement = document.createElement(tagName);
+    newElement.classList.add(className);
+    return newElement;
+}
+
+function displayErrorMessage() {
+    const errorMessage = createNewElements('p', 'error-message');
+    errorMessage.textContent = 'Communication failed on the server side.';
+    ul.appendChild(errorMessage);
+}
+
+async function fetchData() {
+    try {
+        const response = await fetch(endpoint);
+        if (!response.ok) {
+            console.error(`${response.status}:${response.statusText}`);
+        }
+        const result = await response.json();
+        return result.data;
+    } catch (error) {
+        console.error(error);
+    }
+}
+
+
+const renderTabs = (tabElements) => {
+    const fragment = new DocumentFragment();
+    for (const tabElement of tabElements) {
+        const tabList = createNewElements('li', 'tab-list');
```

Comment:
MUST
名前なのですが、ここでようやくtabElementになるのではないでしょうか。引数で受け取っているのはElementではないはずです

Reply:
https://github.com/yoshikawa-akr/js-lessons/pull/17/commits/ca858e7eb927e8f1db538a3f05d93ae9697ceb05
You are certainly right.
I think it is the return value of fetchData(), result.data, that is passed as an argument.
I named it this way because it is returned as an array.